src/renderers/dom/shared/DOMPropertyOperations.js

```diff
@@ -90,6 +90,9 @@ var DOMPropertyOperations = {
               if (value === '') {
                 return true;
               }
+              if (value === '' + expected) {
```

Comment:
If the `value` is `null`, and the attribute is the string `"null"` then this should not match expected.